docs/extensibility-http.md

```diff
@@ -0,0 +1,525 @@
+# Extensibility: HTTP-based Devices
+
+See [Extensibility: HTTP protocol](https://github.com/deislabs/akri/issues/85)
+
+This documentation will be completed once the code is working.
+
+## Agent
+
+Revise `./agent/src/protocols/mod.rs`:
+
+```rust
+mod http;
+```
+
+and
+
+```rust
+fn inner_get_discovery_handler(
+    discovery_handler_config: &ProtocolHandler,
+    query: &impl EnvVarQuery,
+) -> Result<Box<dyn DiscoveryHandler + Sync + Send>, Error> {
+    match discovery_handler_config {
+        ProtocolHandler::http(http) => Ok(Box::new(http::HTTPDiscoveryHandler::new(&http))),
+    }
+}
+```
+
+Revise `./agent/src/main.rs`:
+
+```rust
+#[macro_use]
+extern crate failure;
```

Comment:
why is this needed?

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/b97a4f35d81cfbc81732a8ee28d7dbfdb8be21fb